google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionStatementExecutorImpl.java

```diff
@@ -442,4 +450,112 @@ public StatementResult statementShowRPCPriority() {
   public StatementResult statementShowTransactionIsolationLevel() {
     return resultSet("transaction_isolation", "serializable", SHOW_TRANSACTION_ISOLATION_LEVEL);
   }
+
+  private StatementResult getStatementResultFromQueryPlan(QueryPlan qp){
+
+    if(qp == null) return null;
+
+    ArrayList<Struct> list = new ArrayList<>();
+
+    for(PlanNode pn : qp.getPlanNodesList()){
+      list.add(Struct.newBuilder().set("QUERY PLAN").to(pn.getDisplayName()).build());
+
+    }
+
+    ResultSet rs = ResultSets.forRows(
+        Type.struct(StructField.of("QUERY PLAN", Type.string()))
+        , list);
+
+
+    return StatementResultImpl.of(rs);
+  }
+
+  private StatementResult executeStatement(String sql, QueryAnalyzeMode qam){
+    Statement statement = Statement.newBuilder(sql).build();
+    ResultSet rs = getConnection().analyzeQuery(statement, qam);
+    if(rs.getStats() != null){
+      return getStatementResultFromQueryPlan(rs.getStats().getQueryPlan());
+    }
+    throw SpannerExceptionFactory.newSpannerException(ErrorCode.UNAVAILABLE, String.format("Couldn't fetch stats for %s",sql));
+
+  }
+
+  /*
+  * This method supports the EXPLAIN feature with ANALYSE option only.
+  * Any option other than ANALYSE is not supported currently.
+  * If the statement has combination of analyse with any other option
+  * like "explain (analyse, costs) select * from table"
+  * it will throw exception .
+  */
+
+  @Override
+  public StatementResult statementExplain(String sql) {
+
+    if(sql == null){
+      throw SpannerExceptionFactory.newSpannerException(
+          ErrorCode.INVALID_ARGUMENT, String.format("Invalid String with Explain"));
+
+    }
+
+    if(sql.charAt(0) == '('){
+
+      int index = sql.indexOf(')');
+
+      String arr[] = sql.substring(1, index).split("\\s*,\\s*");
+      boolean isAnalyse = false;
+      for(String probableOptions : arr){
+        String arr2[] = probableOptions.split("\\s+");
+        if(arr2.length >= 3){
+          isAnalyse = false;
+          break;
+        }
+        else if(ClientSideStatementExplainExecutor.EXPLAIN_OPTIONS.contains(arr2[0].toLowerCase())){
+          throw SpannerExceptionFactory.newSpannerException(
+              ErrorCode.UNIMPLEMENTED, String.format("%s is not implemented yet", arr2[0]));
+        }
+        else if(arr2[0].equalsIgnoreCase("analyse") || arr2[0].equalsIgnoreCase("analyze")){
+          isAnalyse = true;
+        }
+        else{
+          isAnalyse = false;
+          break;
+        }
+
+        if(arr2.length == 2){
+          if(arr2[1].equalsIgnoreCase("false")){
+            isAnalyse = false;
+          }
+          else if(!arr2[1].equalsIgnoreCase("true")){
```

Comment:
True may also be written as `1` and `on`.

Any invalid boolean value should cause the method to throw an exception.

Reply:
If there will be an invalid boolean like "explain (analyse foo) select * from table", then it will send "(analyse foo) select * from table" as the query, which will eventually cause exception. Its just the error message will not be appropriate (it will be something like "(analyse foo) select * from table" is not a query).

If possible, can we take up the task of distinguishing the error message sometime after this initial PR is merged?